outlines/generate/__init__.py

```diff
@@ -70,10 +70,10 @@ def __call__(self, prompt, **inference_kwargs):
 
 
 def Generator(
-    model: Union[LocalModel, APIModel],
+    model: Model,
     output_type: Optional[Union[Json, List, Choice, Regex]] = None,
 ):
-    if isinstance(model, APIModel):  # type: ignore
+    if model.model_type == "api":  # type: ignore
```

Comment:
The `# type: ignore` mypy instruction seems no longer required, right? The two dataclasses start to look pretty similar, is the strategy to merge them into one?

Reply:
Yes, but I'm going to revert that change in the end anyway.